blockchain-albatross/src/blockchain.rs

```diff
@@ -592,6 +599,44 @@ impl Blockchain {
                 return Err(PushError::InvalidBlock(BlockError::InvalidJustification));
             }
 
+            // Check if there are two blocks in the same slot. Since we already
+            // verified the validator for the current slot, this is enough to
+            // check for fork proofs. A fork proof doesn't apply to the genesis
+            // block, so skip block 0.
+            if block.block_number() != 0 {
+                // Count the microblocks after the last macroblock
+                let microblocks: Vec<Block> = self.chain_store.get_blocks_at(block.block_number(), false, Some(&read_txn));
+
+                // Get the microheader from the block
+                let microheader1 = block.clone().unwrap_micro().header;
+
+                // Get the justification for the block. We asume that the
+                // validator's signature is valid.
+                let justification1 = block.clone().unwrap_micro().justification.signature;
+
+                // Get the view number from the block
+                let view_number = block.clone().view_number();
```

Comment:
Cloning is a heavy operation that we should avoid as much as possible.
```suggestion
                let microheader1 = &micro_block.header;

                // Get the justification for the block. We asume that the
                // validator's signature is valid.
                let justification1 = &micro_block.justification.signature;

                // Get the view number from the block
                let view_number = micro_block.header.view_number;
```